src/ccl_power.c

```diff
@@ -1238,116 +1135,150 @@ static void ccl_cosmology_compute_power_emu(ccl_cosmology * cosmo, int * status)
     if(fabs(cosmo->params.N_nu_rel - 3.04)>1.e-6){
       *status=CCL_ERROR_INCONSISTENT;
       ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_emu(): Set Neff = 3.04 for cosmic emulator predictions in absence of massive neutrinos.\n");
+      return;
     }
   }
+  double w0wacomb = -cosmo->params.w0 - cosmo->params.wa;
+  if(w0wacomb<8.1e-3){ //0.3^4
+    *status=CCL_ERROR_INCONSISTENT;
+    ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_emu(): w0 and wa do not satisfy the emulator bound\n");
+    return;
+  }
+  if(cosmo->params.Omega_n_mass*cosmo->params.h*cosmo->params.h>0.01){
+    *status=CCL_ERROR_INCONSISTENT;
+    ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_emu(): Omega_nu does not satisfy the emulator bound\n");
+    return;
+  }
 
-  if(!*status){
-    // Prepare to run CLASS for linear scales
-    ccl_fill_class_parameters(cosmo,&fc,parser_length, status);
+  // Check to see if sigma8 was defined
+  if(isnan(cosmo->params.sigma8)){
+    *status=CCL_ERROR_INCONSISTENT;
+    ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_emu(): sigma8 is not defined; specify sigma8 instead of A_s\n");
+    return;
   }
-  
-  if (!*status){
+
+  // Prepare to run CLASS for linear scales
+  ccl_fill_class_parameters(cosmo,&fc,parser_length, status);
+
+  if (*status != CCL_ERROR_CLASS)
```

Comment:
Random question. Does the emulator not have a linear prediction to go with it?

Reply:
Nope, we use CLASS